org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandlerTest.java

```diff
@@ -2643,7 +2643,11 @@ public void testStarImports() throws Exception {
 			"}\n");
 			//@formatter:on
 			int[] loc = findCompletionLocation(unit, "new ArrayL");
+			Job.getJobManager().join(DocumentLifeCycleHandler.DOCUMENT_LIFE_CYCLE_JOBS, new NullProgressMonitor());
 			CompletionList list = server.completion(JsonMessageHelper.getParams(createCompletionRequest(unit, loc[0], loc[1]))).join().getRight();
+			if (list.isIncomplete()) {
```

Comment:
That's a really ugly band-aid. Have you checked the PR without it?

Reply:
checking...

Reply:
I have started 12 jenkins builds.
- 6 builds without the PR - 1 fails
- 6 builds with the PR - 0 fails

The issue happens because of the completion timeout - https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandler.java#L132

Reply:
Are we waiting for the indexing to finish? Could we ensure it's done before running the completion tests? 
Also the tests could control the timeout value with a system property

Reply:
> Are we waiting for the indexing to finish?

Yes, we are. See https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/AbstractProjectsManagerBasedTest.java#L231

> Also the tests could control the timeout value with a system property

The timeout value is hardcoded - https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandler.java#L132

Reply:
Yes so it needs to be changed to something like 

private static long TIMEOUT  = Long.getLong("completion.timeout", 5000);

Reply:
Done.